torchaudio/datasets/folder.py

```diff
@@ -0,0 +1,73 @@
+import os
+from typing import Tuple
+import torchaudio
+from torch.utils.data import Dataset
+from torch import Tensor
+from pathlib import Path
+
+def load_audio_item(filepath: str, path: str, label_name: str) -> Tuple[Tensor, int, str, str]:
```

Comment:
Making this into a plain function makes it hard to customize the loading behavior.

Imagine use-cases like fixed-length portion is required (say 5 seconds chunk), and the audio data in file system are long (like one hour). Without customization, one has to load (decode) the entire audio and they are immediately discarded. This is inefficient.

Not being able to customize the load behavior reduces the value of having a generic dataset like this.

Reply:
Let me look into this. Thanks!

Reply:
Before continue working on this, please open [an issue with Feature Request template](https://github.com/pytorch/audio/issues/new?assignees=&labels=&template=feature-request.md) and explain the benefit to have this dataset implementation in torchaudio for the broader audience, and spec it out. You can reference this PR as well, but we do need the goal defined first and clear design principle, otherwise we cannot accept this PR.

Reply:
Will do, thanks :)